examples/ogbn_train.py

```diff
@@ -0,0 +1,266 @@
+    batch_size=batch_size,
+    shuffle=True,
+    num_workers=num_workers,
+    persistent_workers=True,
+)
+
+
+def train(epoch: int) -> Tuple[Tensor, float]:
+    model.train()
+
+    pbar = tqdm(total=split_idx['train'].size(0))
+    pbar.set_description(f'Epoch {epoch:02d}')
+
+    total_loss = total_correct = 0
+    for batch in train_loader:
+        optimizer.zero_grad()
+        out = model(batch.x, batch.edge_index.to(device))[:batch.batch_size]
+        y = batch.y[:batch.batch_size].squeeze().to(torch.long)
+        loss = F.cross_entropy(out, y)
+        loss.backward()
+        optimizer.step()
+
+        total_loss += float(loss)
+        total_correct += int(out.argmax(dim=-1).eq(y).sum())
+        pbar.update(batch.batch_size)
+
+    pbar.close()
+    loss = total_loss / len(train_loader)
+    approx_acc = total_correct / split_idx['train'].size(0)
+    return loss, approx_acc
+
+
+@torch.no_grad()
+def test(loader: NeighborLoader, val_steps=None) -> float:
+    model.eval()
+
+    total_correct = total_examples = 0
+    for i, batch in enumerate(loader):
+        if val_steps is not None and i >= val_steps:
+            break
+        batch = batch.to(device)
+        batch_size = batch.num_sampled_nodes[0]
+        out = model(batch.x, batch.edge_index)[:batch_size]
+        pred = out.argmax(dim=-1)
+        y = batch.y[:batch_size].view(-1).to(torch.long)
+
+        total_correct += int((pred == y).sum())
+        total_examples += y.size(0)
+
+    return total_correct / total_examples
+
+
+if args.use_gat:
+    from torch_geometric.nn.models import GAT
+    model = GAT(
+        in_channels=dataset.num_features,
+        hidden_channels=num_hidden_channels,
+        num_layers=num_layers,
+        out_channels=dataset.num_classes,
+        dropout=args.dropout,
+        heads=args.num_heads,
+    )
+else:
+    from torch_geometric.nn.models import GraphSAGE
+    model = GraphSAGE(
+        in_channels=dataset.num_features,
+        hidden_channels=num_hidden_channels,
+        num_layers=num_layers,
+        out_channels=dataset.num_classes,
+        dropout=args.dropout,
+    )
+
+model = model.to(device)
+optimizer = torch.optim.Adam(model.parameters(), lr=args.lr,
+                             weight_decay=args.wd)
+
+if verbose:
+    prep_time = round(time.perf_counter() - wall_clock_start, 2)
+    print("Total time before training begins (prep_time)=", prep_time,
+          "seconds")
+    print("Training...")
+
+test_accs = []
+val_accs = []
+times = []
+train_times = []
+inference_times = []
+best_val = best_test = 0.
+start = time.time()
+
+model.reset_parameters()
+optimizer = torch.optim.Adam(model.parameters(), lr=args.lr)
+
+for epoch in range(1, num_epochs + 1):
+    train_start = time.time()
+    loss, _ = train(epoch)
+    train_end = time.time()
+    train_times.append(train_end - train_start)
+
+    inference_start = time.time()
+    val_acc = test(val_loader)
+    test_acc = test(test_loader)
```

Comment:
I don't understand why we need to evaluate the model on the test set every epoch.

Reply:
agreed, @guanxingithub please make the test only happen at the end of all epochs of the training

Reply:
If you agree, I can put all of these extra testing and evaluations under verbose mode. So, from user point of view, they don't need to make these further testing and evaluations. But for someone who wants to make performance and accuracy analysis, she/he can turn on verbose mode to collect raw data of performance and accuracy for further statistical analysis.

Reply:
no you should just not be running test during training, all other metrics are fine. please adjust accordingly

Reply:
If we keep evaluation results on test set in each epoch, that will be very helpful to make further analysis. I did the same evaluation on the test set on other tools. If we can do the same evaluation in PyG, that will be very helpful to compare the accuracy between PyG and other tools apple to apple. Thanks.

Reply:
> that will be very helpful to make further analysis.

What kind of analysis are you referring to specifically?

Reply:
For example, for platform1, and platform2, we collected all these performance and accuracy data, and then we can analyze these data obtained from both platforms, to make a judgment like the hypothesis: whether the performance is same or different, accuracy is difference or not from statistical point of view. If it is significant difference, what could cause the difference. This will build the foundation of our further investigation and improvement. We have done a lot of this kind of analysis not only in PyG, but also other tools.

Reply:
Thanks a lot for elaborating on this!

(I'm still not convinced why evaluating on the test helps because there's no distinction between val and test sets in this script at the moment, and I still believe evaluating the model on the validation set should suffice your needs, but) It sounds like you want to run evaluation on the test set during training for library development for parity against other libraries, but this example script is for general PyG users, but not for PyG developers.

If you agree on this, let's remove the test evaluation from the training loop and just do so at the end of the script once.

Reply:
For example, powerful machine in some big companies. (Please allow me not to disclose the names). Thanks.

Reply:
Based on your instructions, I have updated the code:
1. remove the evaluation on test set during training
2. remove verbose flag